modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -843,6 +971,297 @@ static void run_arithm_s(DST out[], const SRC in[], int width, int chan,
+    int length = width * 3;
+
+    if (std::is_same<T, uchar>::value)
+    {
+        constexpr int nlanes = static_cast<int>(v_uint8::nlanes);
+
+        if (length < 3 * nlanes)
+            return 0;
+
+        int x = 0;
+        v_float32 vectors[12];
+        for (;;)
+        {
+            for (; x <= length - 3 * nlanes; x += 3 * nlanes)
+            {
+                for (int i = 0; i < 12; ++i)
+                {
+                    vectors[i] = v_setr(reinterpret_cast<const uchar*>(in + x + i * nlanes / 4));
+                }
+
+                vx_store(reinterpret_cast<uchar*>(&out[x]), v_pack_u(v_pack(v_round(v_absdiff(vectors[0], s1)),
+                                                                            v_round(v_absdiff(vectors[1], s2))),
+                                                                     v_pack(v_round(v_absdiff(vectors[2], s3)),
+                                                                            v_round(v_absdiff(vectors[3], s1)))));
+
+                vx_store(reinterpret_cast<uchar*>(&out[x + nlanes]), v_pack_u(v_pack(v_round(v_absdiff(vectors[4], s2)),
+                                                                                     v_round(v_absdiff(vectors[5], s3))),
+                                                                              v_pack(v_round(v_absdiff(vectors[6], s1)),
+                                                                                     v_round(v_absdiff(vectors[7], s2)))));
+
+                vx_store(reinterpret_cast<uchar*>(&out[x + 2 * nlanes]), v_pack_u(v_pack(v_round(v_absdiff(vectors[8], s3)),
+                                                                                         v_round(v_absdiff(vectors[9], s1))),
+                                                                                  v_pack(v_round(v_absdiff(vectors[10], s2)),
+                                                                                         v_round(v_absdiff(vectors[11], s3)))));
+            }
+
+            if (x < length && (in != out))
+            {
+                x = length - 3 * nlanes;
+                continue;  // process one more time (unaligned tail)
+            }
+            break;
+        }
+        return x;
+    }
+    else if (std::is_same<T, ushort>::value)
+    {
+        return absdiffc_simd_shorts_c3(reinterpret_cast<const ushort*>(in),
+                                       reinterpret_cast<ushort*>(out), s1, s2, s3, length);
+    }
+    else if (std::is_same<T, short>::value)
+    {
+        return absdiffc_simd_shorts_c3(reinterpret_cast<const short*>(in),
+                                       reinterpret_cast<short*>(out), s1, s2, s3, length);
+    }
+    return 0;
+}
+
+template<typename T>
+CV_ALWAYS_INLINE int absdiffc_simd_c4(const T in[], const float scalar[],
+                                      T out[], int width)
+{
+    v_float32 s = v_set_scalar(scalar, 4);
```

Comment:
As far as I can see second parameter of `v_set_scalar` is known at compile time in all calls. So you can make this function a template or simply add four functions to eliminate a runtime switch inside

Reply:
I did as I understood. Please check.